File: 2024/Leetcode/problem412.py
class Solution:
    def fizzBuzz(self, n: int) -> List[str]:
        answer = []
        specialNums = {3: "Fizz", 5: "Buzz"}

        for num in range(1, n + 1):
            curr = ""
            for val in specialNums:
                if num % val == 0:
                    curr += specialNums[val]

            if curr == "":
                curr = str(num)
            answer.append(curr)

        return answer

File: 2024/Leetcode/problem412.js
/**
 * @param {number} n
 * @return {string[]}
 */
var fizzBuzz = function(n) {
    const getFizzBuzz = (num) => {
        let phrases = {3: 'Fizz', 5: 'Buzz'};

        let result = '';
        for (let dividend in phrases) {
            if (num % dividend === 0) {
                result += phrases[dividend];
            }
        }

        if (result === '') {
            result = num.toString();
        }
    
        return result;
    }

    const arr = [];
    for (let i = 1; i <= n; i++) {
        arr.push(getFizzBuzz(i));
    }

    return arr;
};